packages/tasks/src/model-data.ts

```diff
@@ -66,6 +66,16 @@ export interface ModelData {
 			base_model_name_or_path?: string;
 			task_type?: string;
 		};
+		keras_hub_task_json?: {
+			class_name: string;
+			alt_class_names?: string[];
+		};
+		keras_hub_config_json?: {
+			class_name: string;
+		};
+		keras_hub_tokenizer_json?: {
+			class_name: string;
+		};
```

Comment:
slight preference for a more concise

```suggestion
                keras_hub?: {
                	// relevant task.json content
                };
```

From what I understand, `task.json` is the future-proof way of getting this info correctly. And getting things from `config.json`/`tokenizer.json` is more of a default for previous models up to now. Is my assumption correct or not? If that's the case, then let's focus on parsing only `task.json` to only promote the "correct" way. 

In any case (no matter if the config comes from task.json, config.json or tokenizer.json) I think that having a single field with nested values is better rather than exposing 3 different high-level fields related to `keras_hub`.

Reply:
Taking a look at this recent keras-hub model (https://huggingface.co/evandrarf/health-care-gemma2-kagglex/tree/main), I can see that `task.json`, `config.json`, `tokenizers.json` and `preprocessor.json` are all set. And the content of `task.json` is strictly a superset of the other 3. Do you know if other files are kept for backward compatbility?

Reply:
I agree with you that standardizing on a single config file would be the best. Let me ask the keras team?

Reply:
Matt on the Keras team responded. `task.json` is not always present and that is by design and not a legacy thing. I recommend we deploy the currently implemented logic while we continue the discussion with Matt and possibly simplify.

Reply:
yes and imo we can influence the standardization by supporting the simpler / single-version version that Wauplin mentions